src/main/java/com/netflix/imflibrary/st2067_203/IMFMGASADMConstraintsChecker.java

```diff
@@ -0,0 +1,265 @@
+                            imfTrackFileResourceType.getSourceEncoding(), domNodeObjectModel.getLocalName()));
+                }
+
+                if (((imfTrackFileResourceType.getEditRate().getNumerator()*compositionEditRate.getDenominator()) % (imfTrackFileResourceType.getEditRate().getDenominator()*compositionEditRate.getNumerator()) != 0)) {
+                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CORE_CONSTRAINTS_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL, String.format("The EditRate %s/%s of resource %s is not a multiple of the EditRate of the Main Image Virtual Track %s/%s",
+                            imfTrackFileResourceType.getEditRate().getNumerator(), imfTrackFileResourceType.getEditRate().getDenominator(), imfTrackFileResourceType.getId(), compositionEditRate.getNumerator(), compositionEditRate.getDenominator()));
+                }
+
+                if (!domNodeObjectModel.getFieldAsUL("ContainerFormat").equals(MXF_GC_CLIP_WRAPPED_MGA)) {
+                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CORE_CONSTRAINTS_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL, String.format("EssenceDescriptor ID %s referenced by " +
+                                    "an MGA S-ADM VirtualTrack Resource does not use the correct Essence Container UL: %s vs. %s",
+                            imfTrackFileResourceType.getSourceEncoding(), domNodeObjectModel.getFieldAsUL("ContainerFormat"), MXF_GC_CLIP_WRAPPED_MGA));
+                }
+
+                if (!domNodeObjectModel.getFieldAsUL("SoundCompression").equals(MGA_AUDIO_ESSENCE_UNCOMPRESSED_SOUND_CODING)) {
+                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CORE_CONSTRAINTS_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL, String.format("EssenceDescriptor ID %s referenced by " +
+                                    "an MGA S-ADM VirtualTrack Resource does not use the correct Sound Compression UL: %s vs. %s",
+                            imfTrackFileResourceType.getSourceEncoding(), domNodeObjectModel.getFieldAsUL("SoundCompression"), MGA_AUDIO_ESSENCE_UNCOMPRESSED_SOUND_CODING));
+                }
+
+                if (domNodeObjectModel.getFieldAsInteger("QuantizationBits") != MGASADM_BIT_DEPTH.intValue()) {
+                    imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CORE_CONSTRAINTS_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL, String.format("EssenceDescriptor ID %s referenced by " +
+                                    "an MGA S-ADM VirtualTrack Resource has invalid QuantizationBits field: %s vs. %s",
+                            imfTrackFileResourceType.getSourceEncoding(), domNodeObjectModel.getFieldAsInteger("QuantizationBits"), MGASADM_BIT_DEPTH));
+                }
+
+                if (domNodeObjectModel.getFieldAsString("AudioSampleRate") == null) {
```

Comment:
Recommend also verifying that AudioSampleRate is either 48000 or 96000, as per ST 2067-203 Table 1.